kroxylicious-filters/kroxylicious-record-encryption/src/test/java/io/kroxylicious/filter/encryption/WrapperVersionTest.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import org.assertj.core.api.ThrowableAssert;
+import org.junit.jupiter.api.Test;
+
+import static io.kroxylicious.filter.encryption.WrapperVersion.V1_UNSUPPORTED;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class WrapperVersionTest {
+
+    @Test
+    void unsupportedWrapperVersionThrowsOnUsage() {
+        assertUnsupported(() -> V1_UNSUPPORTED.writeWrapper(null, null, null, 1, null, null, null, null, null, null, null));
+        assertUnsupported(() -> V1_UNSUPPORTED.read(null, null, 1, null, null, null, null, null));
+        assertUnsupported(() -> V1_UNSUPPORTED.readSpecAndEdek(null, null, null));
+    }
+
+    private void assertUnsupported(ThrowableAssert.ThrowingCallable op) {
+        assertThatThrownBy(op).isInstanceOf(EncryptionException.class).hasMessage("unsupported wrapper version");
```

Comment:
Is `unsupported wrapper version` wrapper version going to mean anything to an operator reading the logs?

Reply:
As above, the logs an operator sees should be an exception about the EnvelopeEncryption version from the header being unsupported by this filter version, it shouldn't get as far as trying to read the wrapper.